src/rpp/mutex.cpp

```diff
@@ -75,5 +83,160 @@ namespace rpp
         if (CRITICAL_SECTION* cs = GET_RMUTEX())
             LeaveCriticalSection(cs);
     }
+#elif RPP_FREERTOS
+    /////////////////////////////////////////////////////////////////
+
+    mutex::mutex()
+    {
+        mtx.ctx = xSemaphoreCreateBinary(); // binary semaphore is faster than mutex semaphore
+        xSemaphoreGive((SemaphoreHandle_t)mtx.ctx); // Initialize to available
+    }
+
+    mutex::~mutex()
+    {
+        if (mtx.ctx)
+        {
+            vSemaphoreDelete((SemaphoreHandle_t)mtx.ctx);
+            mtx.ctx = nullptr;
+        }
+    }
+
+    bool mutex::try_lock()
+    { 
+        if (xPortIsInsideInterrupt())
+        {
+            BaseType_t higherPriorityTaskWoken = pdFALSE;
+            bool r = xSemaphoreTakeFromISR((SemaphoreHandle_t)mtx.ctx, &higherPriorityTaskWoken) == pdTRUE;
+            portYIELD_FROM_ISR(higherPriorityTaskWoken);
+            return r;
+        }
+
+        return xSemaphoreTake((SemaphoreHandle_t)mtx.ctx, 0) == pdTRUE;
+    }
+    
+    void mutex::lock()
+    {
+        if (xPortIsInsideInterrupt())
+        {
+            BaseType_t higherPriorityTaskWoken = pdFALSE;
+            xSemaphoreTakeFromISR((SemaphoreHandle_t)mtx.ctx, &higherPriorityTaskWoken);
+            portYIELD_FROM_ISR(higherPriorityTaskWoken);
+        }
+
+        else
+            xSemaphoreTake((SemaphoreHandle_t)mtx.ctx, portMAX_DELAY);
+    }
+    
+    void mutex::unlock()
+    {
+        if (xPortIsInsideInterrupt())
+        {
+            BaseType_t higherPriorityTaskWoken = pdFALSE;
+            xSemaphoreGiveFromISR((SemaphoreHandle_t)mtx.ctx, &higherPriorityTaskWoken);
+            portYIELD_FROM_ISR(higherPriorityTaskWoken);
+        }
+
+        else
+            xSemaphoreGive((SemaphoreHandle_t)mtx.ctx);
```

Comment:
Here as well, this space between } and else is not good for readability/maintainability